galter_subjects_utils/contrib/lcsh/converter.py

```diff
@@ -115,25 +135,47 @@ def find_explanation(nodes_of_change_note):
     return ""
 
 
+pattern = r"covered by .+ subject heading( {(?P<subject_new>.+)})? .+(?P<id_new>sh\s?\d+)$"  # noqa
+regex_for_replacement = re.compile(pattern)
```

Comment:
Why are these outside of a function? Is this being defined as a constant?

Reply:
These are outside the function to take advantage of regular expression compilation. Basically, `re.compile(pattern)` is a one-time hit in exchange for an efficient regular expression object that is more performant at executing the `search` operations than `re.search(<pattern>, <string>)`. *Technically* Python does regex compilation under the hood of `re.search` and caches the result such that this code pattern shouldn't matter if we are using few regexes. But given the size of the code base, I am assuming we have nothing to lose by performing this compilation.

Python also doesn't have enforced constants per se, but it would be the norm to capitalize these module level variables to indicate that they shouldn't be changed. I didn't here, because I didn't want them to standout/be seen as regular "constants" that could be reused elsewhere. Not very attached to that though, so I could change it if it's important for others.

Reply:
Thanks for the explanation! I think it's fine as is. Just wanted to make sure I understood what was going on.